contracts/JetStaking/JetStakingV3.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "./JetStakingV2.sol";
+
+/**
+ * @title JetStakingV3
+ * @author Aurora Team
+ *
+ * @dev Implementation of Jet staking contract
+ *
+ *      This contract implements the staking mechanics for AURORA ERC20 token.
+ *      A user can stake any amount of AURORA tokens, and get rewarded in both
+ *      AURORA and other stream tokens based on the rewards schedules.
+ *      Stream rewards can be claimed any time however AURORA can't be claimed
+ *      unless the user unstakes his full/partial amount of shares.
+ *
+ *      This contract is AdminControlled which has a tremendous power. However
+ *      hopfully it be governed by a community wallet.
+ */
+contract JetStakingV3 is JetStakingV2 {
+    // RPS_MULTIPLIER = Aurora_max_supply x weight(1000) * 10 (large enough to always release rewards) =
+    // 10**9 * 10**18 * 10**3 * 10= 10**31
+    uint256 private constant RPS_MULTIPLIER = 1e31;
+    // we store all the current streams (old streams)
+    // in this state variable. The main goal is to avoid
+    // mixing old weighted shares stream calculations and the new ones.
+    mapping(uint256 => bool) public oldWeightedShareStreams;
+
+    /// @dev set the old weighted share streams
+    /// @notice This function must be called prior upgrading the contract.
+    /// And the contract must be paused before initializing it.
+    function initializeOldWeightedStreams()
+        public
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        require(paused != 0, "E16"); // REQUIRE_PAUSE
+        for (uint256 i = 1; i < streams.length; i++) {
+            oldWeightedShareStreams[i] = true;
+        }
+    }
+
+    /// @dev calculates and gets the latest reward per share (RPS) for a stream
+    /// @param streamId stream index
+    /// @return streams[streamId].rps + scheduled reward up till now
+    function getLatestRewardPerShare(uint256 streamId)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        require(streamId != 0, "E29"); // AURORA_REWARDS_COMPOUND
+        uint256 totalShares = oldWeightedShareStreams[streamId]
```

Comment:
Since stream ids are sequential, wouldn't it be simpler to check `streamId >= 5` directly instead of using `oldWeightedShareStreams` and `initializeOldWeightedStreams`